### PR TITLE
Emma/generate methods

### PR DESCRIPTION
**Pull request by edix9**

### CODE REVIEW

File: src/Criterion.cpp [hunk 1]
```diff
@@ -24,6 +24,19 @@ Criterion::Criterion(const Criterion &crit){
...
+    else { direction_ = 1; }
+}
+
+void Criterion::generateWeight(unsigned long int seed){
+    srand(seed);
+    weight_ = (float) rand() / RAND_MAX ;
+}
+
+
+
```
Comment (thibmonsel) [inline src/Criterion.cpp:39]: Should we have one extra function that combines both in order to generate a random Criterion with one method function call ?

Comment (Mostah) [reply src/Criterion.cpp:39]: We could implement such functions indeed, maybe let's wait to analyze the different algorithms to check which initializations are we going to use?

File: src/Criterion.h [hunk 2]
```diff
@@ -37,6 +37,21 @@ class Criterion : public AtomicMCDAObject  {
     ~Criterion();
     friend std::ostream & operator <<( std::ostream & out, const Criterion & crit );
 
+    /**
+     * generateDirection generator of random direction
+     * can also be used as reinitialisation for direction
+     *
```
Comment (Mostah) [inline src/Criterion.h]: nit: for the doc I think it's better if we add the params in its description (with @param)

File: src/Criterion.h [hunk 3]
```diff
@@ -37,6 +37,21 @@ class Criterion : public AtomicMCDAObject  {
...
+    /**
+     * generateDirection generator of random direction
+     * can also be used as reinitialisation for direction
+     * 
+     */
+    void generateDirection(unsigned long int seed = time(NULL));
+
+    /**
+     * generateWeight generator of random weight
+     * can also be used as reinitialisation for weight
```
Comment (Mostah) [inline src/Criterion.h]: nit: same here

File: src/Criterion.cpp (hunk 1 above)
Comment (thibmonsel) [reply src/Criterion.cpp:39]: That function would be useful for the dataGenerator im working on for example 👍

Comment (thibmonsel) [reply src/Criterion.cpp:39]: and where we could control the randomness with a seed in order to make test work

File: test/TestCriterion.cpp [hunk 4]
```diff
@@ -36,6 +36,78 @@ TEST(TestCriterion, TestConstructorByCopy) {
...
+  Criterion crit2 = Criterion(id2);
+  unsigned long int seed2 = 40;
+  // generating a direction to compare to the one generated in generateDirection
+  // should be the same since we are using the same seed 
+  srand(seed);
+  int direction = 1;
+  if ( ((float) rand() / RAND_MAX) < 0.5){ direction = -1; }
+  crit.generateDirection(seed);
+
+  srand(seed2);
```
Comment (Mostah) [inline test/TestCriterion.cpp]: It looks like you are testing twice the same feature here

File: test/TestCriterion.cpp [hunk 5]
```diff
@@ -36,6 +36,78 @@ TEST(TestCriterion, TestConstructorByCopy) {
...
+  // should be the same since we are using the same seed 
+  srand(seed); 
+  float weight = ((float) rand() / RAND_MAX);
+  crit.generateWeight(seed);
+
+  srand(seed2); 
+  float weight2 = ((float) rand() / RAND_MAX);
+  crit2.generateWeight(seed2);
+
+  float weight3 = ((float) rand() / RAND_MAX);
```
Comment (Mostah) [inline test/TestCriterion.cpp]: Looks like you could remove `weight3` as you are not using it

File: test/TestCriterion.cpp [hunk 6]
```diff
@@ -36,6 +36,78 @@ TEST(TestCriterion, TestConstructorByCopy) {
...
+  
+  EXPECT_FLOAT_EQ(weight, crit.getWeight());
+  EXPECT_FLOAT_EQ(weight2, crit2.getWeight());
```
Comment (Mostah) [inline test/TestCriterion.cpp]: same comment as for the test above, looks like you are testing twice the same feature

**Review by Mostah — APPROVED** [review]
A few comments to clean up a bit the code, otherwise it looks good to me 👍

File: src/Criterion.cpp (hunk 1 above)
Comment (edix9) [reply src/Criterion.cpp:39]: will be solved in Thibault's generator

File: src/Criterion.h (hunk 3 above)
Comment (edix9) [reply src/Criterion.h]: corrected

File: src/Criterion.h (hunk 2 above)
Comment (edix9) [reply src/Criterion.h]: corrected

File: test/TestCriterion.cpp (hunk 5 above)
Comment (edix9) [reply test/TestCriterion.cpp]: removed

File: test/TestCriterion.cpp (hunk 4 above)
Comment (edix9) [reply test/TestCriterion.cpp]: Right, I just wanted to test with different seeds but not really useful. Will remove

**Pull request merged**